posydon/binary_evol/MESA/step_mesa.py

```diff
@@ -1251,29 +1259,39 @@ def __call__(self, binary):
         state_1 = self.binary.star_1.state
         state_2 = self.binary.star_2.state
         event = self.binary.event
-        mass_ratio = self.binary.star_2.mass/self.binary.star_1.mass
+        m1 = self.binary.star_1.mass
+        m2 = self.binary.star_2.mass
+        mass_ratio = m2/m1
         p = self.binary.orbital_period
-        p_max_HMS_grid = 6500
-        if (state_1 == 'H-rich_Core_H_burning'
-                and state_2 == 'H-rich_Core_H_burning' and event == 'ZAMS'
-                and mass_ratio <= 1. and p <= p_max_HMS_grid):
+        if (state_1 == 'H-rich_Core_H_burning' and 
+            state_2 == 'H-rich_Core_H_burning' and 
+            event == 'ZAMS' and
+            self.m1_min <= m1 <= self.m1_max and 
+            self.q_min <= mass_ratio <= self.q_max and 
+            self.p_min <= p <= self.p_max):
             self.flip_stars_before_step = False
             super().__call__(self.binary)
-        elif (state_1 == 'H-rich_Core_H_burning'
-              and state_2 == 'H-rich_Core_H_burning'
-              and event == 'ZAMS' and mass_ratio > 1. and p <= p_max_HMS_grid):
+        elif (state_1 == 'H-rich_Core_H_burning' and 
+              state_2 == 'H-rich_Core_H_burning' and 
+              event == 'ZAMS' and
+              self.m1_min <= m2 <= self.m1_max and
+              1./self.q_min >= mass_ratio > self.q_max and 
+              self.p_min <= p <= self.p_max):
             self.flip_stars_before_step = True
             super().__call__(self.binary)
-        elif (state_1 == 'H-rich_Core_H_burning'
-              and state_2 == 'H-rich_Core_H_burning'
-              and event == 'ZAMS'
-              and p > p_max_HMS_grid):              # redirect if outside grid
+        # redirect if outside grid
+        elif (state_1 == 'H-rich_Core_H_burning' and 
+              state_2 == 'H-rich_Core_H_burning' and 
+              event == 'ZAMS' and 
+              p > self.p_max):
             self.binary.event = 'redirect'
             return
-        elif (state_1 == 'H-rich_Central_C_depletion'):     # redirect if CC1
+        # redirect if CC1
+        elif (state_1 == 'H-rich_Central_C_depletion'):     
             self.binary.event = 'CC1'
             return
-        elif (state_2 == 'H-rich_Central_C_depletion'):     # redirect if CC2
+        # redirect if CC2
+        elif (state_2 == 'H-rich_Central_C_depletion'):     
             self.binary.event = 'CC2'
             return
         else:
```

Comment:
Why don't we send stars with masses outside the grid to redirect as well? Our single star models cover a larger range in masses, especially at the low mass end.

Reply:
I though we used to do this, but now I suspect we did remove this as step_detached would like to do matching with star properties that are not yet defined. Maybe with the redesign of `step_detached`, we can now handle mapping to a single star only with respect to the ZAMS mass when t=0. What do you think @ezapartas? @kasdaglie, this is relevant for your project as well, right now a HMS-HMS binary system outside the maximal orbital period is not redirected to the `step_detached` but in principle, it should.